modules/dicom_archive/php/dicom_archive.class.inc

```diff
@@ -98,6 +100,65 @@ class Dicom_Archive extends \NDB_Menu_Filter
     }
 
 
+    /**
+     * Handle a PSR7 Request for the dicom_archive endpoint. Also calls a
+     * validation function to ensure necessary configuration settings exist.
+     *
+     * @param ServerRequestInterface $request The PSR15 Request being handled
+     *
+     * @return ResponseInterface The PSR15 response for the page.
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        try {
+            $this->validateConfig();
+        } catch (\ConfigurationException $e) {
+            error_log($e);
+            return (new \Zend\Diactoros\Response())
+                ->withStatus(500, 'Forbidden')
+                ->withBody(
+                    new \LORIS\Http\StringStream(
+                        'Required configuration settings for DICOM ' .
+                        'anonymization are missing. Cannot continue.'
+                    )
+                );
+        }
+        return parent::handle($request);
+    }
+
+    /**
+     * Checks that the required configuration settings are properly set. Throws
+     * a ConfigurationException if not.
+     *
+     * @return void
+     *
+     * @throws \ConfigurationException
+     */
+    function validateConfig(): void
+    {
+        /* Ensure that all required configuraton settings are properly set
+         * before continuing.
+         */
+        $config        = \NDB_Config::singleton()->getSetting("imaging_modules");
+        $configOptions = array(
+                          'patientNameRegex',
+                          'LegoPhantomRegex',
+                          'LivingPhantomRegex',
+                          'patientIDRegex',
+                         );
```

Comment:
I don't see any reference to these in this file. I only see them referenced from the view session page, so I don't think this is the right place for this.

Unless I'm missing something, I think this should either:

1. Be moved to the viewsession page, where the configs are used and it just needs to add a !empty check or
2. Be checked by the Module class for the module, not the menu  page's class.

Reply:
Theyr'e used by `dicomarchiveanonymizer.class.inc` (I mentioned this in my description but maybe I wasn't clear). Would you prefer that the checks go there instead?

Alternatively I can make the comment above more clear about where those settings are used.

Reply:
They should checked close to the thing that they're validating.

The DicomArchiveAnonymizer constructor seems reasonable, or in the module's Module class.